File: 0202. Happy Number/Solution.py
class Solution:
    def isHappy(self, n):
        """
        type n: int
        rtype: bool
        """
        maps = {}
        sums = 0
        while (sums != 1):
            sums = 0
            while n:
                digit = n % 10
                sums += digit ** 2
                n = n // 10
            if sums in maps:
                return False
            maps[sums] = sums
            n = sums
        return True

File: 0202. Happy Number/Solution.cpp
#include <unordered_map>

using std::unordered_map;

class Solution {
public:
    bool isHappy(int n) {
        unordered_map<int, int> mpp;
        int sum = 0;
        while (sum != 1)
        {
            sum = 0;
            while (n)
            {
                int digit = n % 10;
                sum += digit * digit;
                n /= 10;
            }
            if (mpp.find(sum) != mpp.end())
                return false;
            mpp[sum] = sum;
            n = sum;
        }
        return true;
    }
};